PlayFabClientSDK/source/ITransportPlugin.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Interface of any transport SDK plugin.
+    /// </summary>
+    public interface ITransportPlugin: IPlayFabPlugin
+    {
+        Task<object> DoPost(string urlPath, object request, Dictionary<string, string> headers);
```

Comment:
Accepting any type of object as the request seems a bit to open ended in my opinion. It seems like we would want to limit to objects which can be serialized (or maybe PlayFabRequestCommon). I have similar thoughts around the return type of the function.

Reply:
I had PlayFabRequestCommon in the initial version of this but Paul convinced me it doesn't have to be. The idea is to allow customers to pass whatever (it doesn't even have to come from our API methods). 

---
In reply to: [201476336](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r201476336) [](ancestors = 201476336)